README.rst

```diff
@@ -0,0 +1,227 @@
+    res <- withSimpleLocalRunner [absdir|/tmp/funflow|] $ \run ->
+      run favouriteNumber ()
+    case res of
+      Left err -> putStrLn $ "Something went wrong: " ++ show err
+      Right (name, num) -> putStrLn $ "Hi, " ++ name ++ ", your favourite number is " ++ num
+
+Defining external tasks
+-----------------------
+
+External tasks are used to run steps outside of the Haskell process.
+Fundamentally, an external task will resolve to a command-line call to another
+application. This can obviously be done with `stepIO`, but using external tasks
+has a couple of big advantages:
+
+- External tasks fit in naturally with the content store framework. Both the
+  inputs to the task and the actual definition of the task will be used to
+  determine the resultant hash, so that if, say, a script changes, the
+  results will be recomputed. This is hard to achieve with `stepIO`.
+- External tasks are naturally distributable. When running in a production setting,
+  you're likely to want to distribute tasks among multiple machines. This is
+  not, in general, achievable with arbitrary `IO` computations.
+
+FunFlow's current approach to external tasks is heavily based on Docker_. Using
+Docker allows tasks to be self-contained, and adds minimal requirements to the
+system being used to host funflow instances (they just need to have docker running!)
+
+To use a docker container as an external step, one simply needs to define a function
+of type `a -> Docker.Config`, where `a` is the input type to the flow. At its core,
+this means:
+
+- Specifying the docker image (and optionally, image ID) to use. For example,
+  `nixos/nix:1.11.14`.
+- Specifying the path to the command which will be run within the container.
+- Specifying which inputs (from the content store) are to be mounted within
+  the container.
+
+Here's an example::
+
+  myDockerStep :: SimpleFlow CS.Item CS.Item
+  myDockerStep = docker $ \input -> Docker.Config
+    { Docker.image = "nixos/nix"
+    , Docker.optImageID = Just "1.11.14"
+    , Docker.input = Docker.SingleInput input
+    , Docker.command = "myScript.sh"
+    , Docker.args =
+        [
+        , "--input_dir", "/input/"
+        , "--output_dir", "/output/"
+        ]
+    }
+
+A `CS.Item` refers to an item within the content store. You can use
+`putInStore`, `getFromStore`, `copyFileToStore` and similar tools to add and
+fetch files from the store. This lets you interleave internal and external
+computations.
+
+Running on multiple machines
+----------------------------
+
+To run on multiple machines, you need to use one of the distributable
+coordinators - either the Redis coordinator or the SQLite coordinator. To do
+this, you need to:
+
+- Start some *executors* pointed at the coordinator.
+- Run your flow using that coordinator.
+
+The simplest way to run an executor is to use the bundled `ffexecutord` executable.
+This can work with either the Redis or SQLite coordinators. Here's how we might
```

Comment:
I'd break the line at "Here's", because it introduces a new idea, so should be a new paragraph.